Apps/phdossier/dossier_connector.py

```diff
@@ -0,0 +1,342 @@
+
+        self._state = None
+
+        # Variable to hold a base_url in case the app makes REST calls
+        # Do note that the app json defines the asset config, so please
+        # modify this as you deem fit.
+        self._base_url = None
+
+    def _make_rest_call(self, endpoint):
+
+        base_url = "https://api.activetrust.net:8000/api/"
+
+        config = self.get_config()
+        api_key = config["api_key"].encode('utf-8')
+
+        headers = {"Content-Type": "application/json"}
+
+        url = "{}{}".format(base_url, endpoint)
+        r = requests.get(url, headers=headers, auth=HTTPBasicAuth(api_key, ""))
+
+        return r.json(), r.status_code
+
+    def _handle_test_connectivity(self, param):
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # NOTE: test connectivity does _NOT_ take any parameters
+        # i.e. the param dictionary passed to this handler will be empty.
+        # Also typically it does not add any data into an action_result either.
+        # The status and progress messages are more important.
+
+        self.save_progress("Connecting to endpoint")
+        # make rest call
+        _, status_code = self._make_rest_call("services/intel/lookup/targets")
+
+        if status_code == 200:
+            self.save_progress("Successfully connected and authenticated")
+            return action_result.set_status(phantom.APP_SUCCESS)
+        else:
+            self.save_progress("Could not connect to service")
+            return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_lookup_domain(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # Access action parameters passed in the 'param' dictionary
+
+        # Required values can be accessed directly
+        domain = param['domain']
+
+        # format our url
+        url = "services/intel/lookup/indicator/host?value={}&source=atp&wait=true".format(domain)
+
+        # make rest call
+        response, status_code = self._make_rest_call(url)
+
+        if status_code == 200:
+            # Add the response into the data section
+            action_result.add_data(response["results"])
+            threat_level = 0
+            threat_confidence = 0
+
+            # this gets the highest theat level and confidence score for the summary
+            for i in response["results"][0]["data"]["threat"]:
+
+                if i["threat_level"] > threat_level:
+                    threat_level = i["threat_level"]
+
+                if "confidence" in i and i["confidence"] > threat_confidence:
+                    print(i['confidence'])
```

Comment:
Either take this out, add it as progress, or add it as debug

Reply:
If I remove the print statement and commented code below, then this IF condition will be useless. I checked against the submitted code and it is added by app dev as is.

Reply:
Then do this:

```suggestion
                    self.debug_print(i['confidence'])
```

Reply:
Fixed.